source/extensions/stat_sinks/hystrix/hystrix.cc

```diff
@@ -0,0 +1,379 @@
+      window_size_(DEFAULT_NUM_OF_BUCKETS + 1) {
+  init();
+}
+
+void HystrixSink::init() {
+  Server::Admin& admin = server_.admin();
+  ENVOY_LOG(debug,
+            "adding hystrix_event_stream endpoint to enable connection to hystrix dashboard");
+  admin.addHandler("/hystrix_event_stream", "send hystrix event stream",
+                   MAKE_ADMIN_HANDLER(handlerHystrixEventStream), false, false);
+}
+
+Http::Code HystrixSink::handlerHystrixEventStream(absl::string_view,
+                                                  Http::HeaderMap& response_headers,
+                                                  Buffer::Instance&,
+                                                  Server::AdminStream& admin_stream) {
+
+  response_headers.insertContentType().value().setReference(
+      Http::Headers::get().ContentTypeValues.TextEventStream);
+  response_headers.insertCacheControl().value().setReference(
+      Http::Headers::get().CacheControlValues.NoCache);
+  response_headers.insertConnection().value().setReference(
+      Http::Headers::get().ConnectionValues.Close);
+  response_headers.insertAccessControlAllowHeaders().value().setReference(
+      Http::Headers::get().AccessControlAllowHeadersValue.AccessControlAllowHeadersHystrix);
+  response_headers.insertAccessControlAllowOrigin().value().setReference(
+      Http::Headers::get().AccessControlAllowOriginValue.All);
+  response_headers.insertNoChunks().value().setReference("0");
+
+  Http::StreamDecoderFilterCallbacks& stream_decoder_filter_callbacks =
+      admin_stream.getDecoderFilterCallbacks();
+
+  registerConnection(&stream_decoder_filter_callbacks);
+
+  admin_stream.setEndStreamOnComplete(false); // set streaming
+
+  // Separated out just so it's easier to understand
+  auto on_destroy_callback = [this, &stream_decoder_filter_callbacks]() {
+    ENVOY_LOG(debug, "stopped sending data to hystrix dashboard on port {}",
+              stream_decoder_filter_callbacks.connection()->remoteAddress()->asString());
+
+    // Unregister the callbacks from the sink so data is no longer encoded through them.
+    unregisterConnection(&stream_decoder_filter_callbacks);
+  };
+
+  // Add the callback to the admin_filter list of callbacks
+  admin_stream.addOnDestroyCallback(std::move(on_destroy_callback));
+
+  ENVOY_LOG(debug, "started sending data to hystrix dashboard on port {}",
+            stream_decoder_filter_callbacks.connection()->remoteAddress()->asString());
+  return Http::Code::OK;
+}
+
+void HystrixSink::flush(Stats::Source&) {
+  if (callbacks_list_.empty())
+    return;
+  incCounter();
+  std::stringstream ss;
+  Upstream::ClusterManager::ClusterInfoMap clusters = server_.clusterManager().clusters();
+  for (auto& cluster : clusters) {
+    Upstream::ClusterInfoConstSharedPtr cluster_info = cluster.second.get().info();
+
+    // update rolling window with cluster stats
+    updateRollingWindowMap(cluster_info);
```

Comment:
nit: why not just do the cache lookup based on cluster name out here and pass the reference into both methods?

Reply:
IIUC - you mean to move the `std::unique_ptr<ClusterStatsCache>& cluster_stats_cache_ptr = cluster_stats_cache_map_[cluster_name];` up to flush()? Please see the new version , is this what you had in mind?